be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -192,8 +192,9 @@ object DbActor extends AskPatternConstants {
   // FIXME: find way to keep track of db size
   sealed case class DbActor(mediatorRef: ActorRef, db: DB, DATABASE_FOLDER: String) extends Actor with ActorLogging {
 
```

Comment:
This file is absolutely correct. However, what do you think about creating a class `Cache`? This way, you could hide the dirty things from DbActor (e.g. line 337 or 262). Moreover, this could be useful if we want to add a cache for something other than LaoData in the future

Open for discussion/suggestion! What do you think @tuomas-p ?

Reply:
This could be a good idea indeed, however, it should be defined as a typed class then (like Map), so I don't know how useful it would be in practice, as all the operations are very basic Map operations, but it could be cleaner

Reply:
To me, for now, the best option would maybe be to add a helper method in DbActor (as it's the only class directly in contact with the database) which verifies the validity of the data in the map instead of having to do all the checks manually (taking a map and a key as arguments in order to be valid for other "caches" too)